nncf/common/accuracy_aware_training/training_loop.py

```diff
@@ -284,67 +331,79 @@ def run(self, model, train_epoch_fn, validate_fn, configure_optimizers_fn=None,
+                                                              compressed_model_accuracy)
+            self.runner.add_tensorboard_scalar('val/accuracy_aware/accuracy_budget', accuracy_budget,
                                                self.runner.cumulative_epoch_count)
+            if self.runner.stop_training(self.compression_controller):
+                nncf_logger.info('Stopped training for the current compression rate due to early stopping criteria.')
+                force_updating_target_compression_rate = True
+
+            self.runner.update_learning_rate()
+
         self.runner.load_best_checkpoint(model)
         compressed_model_accuracy = self.runner.validate(model)
         possible_checkpoint_compression_rates = self.runner.get_compression_rates_with_positive_acc_budget()
         best_checkpoint_compression_rate = max(possible_checkpoint_compression_rates)
-        nncf_logger.info('The final compressed model has {} compression rate with {} accuracy'.format(
-            best_checkpoint_compression_rate, compressed_model_accuracy))
+        nncf_logger.info('The final compressed model has {} compression rate with {} accuracy. '
+                         'The original model accuracy is {})'.format(best_checkpoint_compression_rate,
+                                                                     compressed_model_accuracy,
+                                                                     self.runner.uncompressed_model_accuracy))
         return model
 
-    @staticmethod
-    def _run_initial_training_phase(model, accuracy_aware_controller, runner):
-        runner.configure_optimizers()
-        for _ in range(runner.initial_training_phase_epochs):
-            runner.train_epoch(model, accuracy_aware_controller)
-        compressed_model_accuracy = runner.validate(model)
-        runner.accuracy_bugdet = compressed_model_accuracy - runner.minimal_tolerable_accuracy
-        runner.add_tensorboard_scalar('val/accuracy_aware/accuracy_bugdet',
-                                      runner.accuracy_bugdet, runner.cumulative_epoch_count)
-        nncf_logger.info('Accuracy budget value after training is {}'.format(runner.accuracy_bugdet))
-
-    def _update_target_compression_rate(self, accuracy_aware_controller, runner):
-        current_compression_rate = accuracy_aware_controller.compression_rate
+    def _run_initial_training_phase(self, model):
+        nncf_logger.info('Start the initial training phase')
+
+        maximal_total_epochs = self.runner.maximal_total_epochs
+        self.runner.maximal_total_epochs = self.runner.initial_training_phase_epochs
+        model = self._run_early_exit_training_loop(model)
+        self.runner.maximal_total_epochs = maximal_total_epochs
+
+        nncf_logger.info('Finish the initial learning phase')
+        return model
+
+    def _update_target_compression_rate(self, runner, force_update=False):
```

Comment:
Type hints everywhere

Reply:
It was decided to improve code quality within a separate task